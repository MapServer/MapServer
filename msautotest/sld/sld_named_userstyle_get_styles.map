#
# REQUIRES: INPUT=GDAL OUTPUT=PNG SUPPORTS=WMS

# GetStyles
# RUN_PARMS: sld_named_userstyle_get_styles.xml [MAPSERV] QUERY_STRING="map=[MAPFILE]&SERVICE=WMS&VERSION=1.1.1&REQUEST=GetStyles&LAYERS=lline" > [RESULT]

MAP

NAME TEST
SHAPEPATH ./data
SIZE 300 300
IMAGECOLOR 100 100 100
FONTSET "data/fonts.lst"
PROJECTION
   "init=epsg:4326"
END
WEB
  IMAGEPATH "/tmp/ms_tmp/"
  IMAGEURL "/ms_tmp/"
  METADATA
    "wms_title"		   "Test SLD"
    "wms_onlineresource"   "http://localhost/path/to/wms_simple?"
    "wms_srs"		   "EPSG:4326"
    "ows_schemas_location" "http://schemas.opengis.net"
    "ows_enable_request" "*" 
    "ows_sld_enabled" "true"
  END
END

LAYER
  NAME lline
  DATA hexagon
  TYPE LINE
  STATUS ON
  SIZEUNITS pixels
  CLASSGROUP "red_and_blue"
  CLASS
    GROUP "red_and_blue"
    STYLE
      WIDTH 5
      OPACITY 50
      COLOR 255 0 0
    END
  END
  CLASS
    GROUP "red_and_blue"
    STYLE
      WIDTH 3
      OPACITY 50
      COLOR 0 0 255
    END
  END
  CLASS
    GROUP "green"
    STYLE
      WIDTH 5
      OPACITY 50
      COLOR 0 255 0
    END
  END
END

END
