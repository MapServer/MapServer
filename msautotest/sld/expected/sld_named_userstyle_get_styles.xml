Content-Type: application/vnd.ogc.sld+xml; charset=UTF-8

<StyledLayerDescriptor version="1.0.0" xmlns="http://www.opengis.net/sld" xmlns:gml="http://www.opengis.net/gml" xmlns:ogc="http://www.opengis.net/ogc" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.opengis.net/sld http://schemas.opengis.net/sld/1.0.0/StyledLayerDescriptor.xsd">
<NamedLayer>
<Name>lline</Name>
<UserStyle>
<Name>red_and_blue</Name>
<IsDefault>true</IsDefault>
<FeatureTypeStyle>
<Rule>
<LineSymbolizer>
<Stroke>
<CssParameter name="stroke">#ff0000</CssParameter>
<CssParameter name="stroke-opacity">0.50</CssParameter>
<CssParameter name="stroke-width">5.00</CssParameter>
</Stroke>
</LineSymbolizer>
</Rule>
<Rule>
<LineSymbolizer>
<Stroke>
<CssParameter name="stroke">#0000ff</CssParameter>
<CssParameter name="stroke-opacity">0.50</CssParameter>
<CssParameter name="stroke-width">3.00</CssParameter>
</Stroke>
</LineSymbolizer>
</Rule>
</FeatureTypeStyle>
</UserStyle>
<UserStyle>
<Name>green</Name>
<FeatureTypeStyle>
<Rule>
<LineSymbolizer>
<Stroke>
<CssParameter name="stroke">#00ff00</CssParameter>
<CssParameter name="stroke-opacity">0.50</CssParameter>
<CssParameter name="stroke-width">5.00</CssParameter>
</Stroke>
</LineSymbolizer>
</Rule>
</FeatureTypeStyle>
</UserStyle>
</NamedLayer>
</StyledLayerDescriptor>

